apps/vaporgui/AnnotationEventRouter.cpp

```diff
@@ -68,70 +73,67 @@ std::vector<std::string> split(const std::string &s, char delim)
 
 }    // namespace
 
-AnnotationEventRouter::AnnotationEventRouter(QWidget *parent, ControlExec *ce) : QWidget(parent), Ui_AnnotationGUI(), EventRouter(ce, AnnotationParams::GetClassType())
+AnnotationEventRouter::AnnotationEventRouter(QWidget *parent, ControlExec *ce) : QWidget(parent), EventRouter(ce, AnnotationParams::GetClassType())
 {
-    setupUi(this);
+    setLayout(new QVBoxLayout);
+    _animConnected = false;
+    _ap = NULL;
+
+    VSection *annotationTab = new VSection("Axis Annotations");
+    _axisAnnotationGroup1 = new PGroup({
+        new PCheckbox(AxisAnnotation::_annotationEnabledTag, "Axis Annotations Enabled"),
+        new PCheckbox(AxisAnnotation::_latLonAxesTag, "Annotate with lat/lon"),
+    });
+    annotationTab->layout()->addWidget(_axisAnnotationGroup1);
 
-    _textSizeCombo = new Combo(axisTextSizeEdit, axisTextSizeSlider, true);
-    _digitsCombo = new Combo(axisDigitsEdit, axisDigitsSlider, true);
-    _ticWidthCombo = new Combo(ticWidthEdit, ticWidthSlider);
-    _annotationVaporTable = new VaporTable(axisAnnotationTable);
+    QTableWidget *annotationTable = new QTableWidget;
+    _annotationVaporTable = new VaporTable(annotationTable);
```

Comment:
One of the things that the values in this table need to do is convert meters to lat-lon and back, done with the function convertLonLatToPCS().

The annotations are also stored in the params database as normalized values between 0 and 1, which need to be scaled according to the current domain extents.

The logic in these functions requires the DataStatus class.  I'm not sure how it can be accessed in PWidgets, so I kept this table as-is.

Reply:
The DataStatus is acquired from the ControlExec which is a singleton.

Reply:
Right, but PWidgets don't have access to the ControlExec, or DataStatus.  Am I mistaken?

Reply:
- Since these are specialized widgets that require extra information, you could create a specialized PWidget that has access to this extra information. This is greatly simplified by that extra information being available through a singleton.
- This widget takes the world coordinates and converts them to normalized coordinates which are saved in the params database. The renderer then takes these normalized coordinates and converts them to world coordinates to be rendered. Could this by simplified by simply keeping them as world coordinates, removing the need for the extra information and allowing the use of generic PWidgets?

Reply:
Regarding your first note: 
I thought about this too but I believe it involves several new classes, like a PGroup that can pass the ControlExec onto PLineEdits; some of which would need the ControlExec (for the minTics parameter) and other PLineEdits that do not (for the numTics parameter).  I felt that it was suitable to use our GUI rule #3, where in certain situations it's ok to use VWidgets along with classical signals and slots.

Regarding your second note: 
As I recall, there are two use cases for normalizing the annotation coordinates.  @clyne let me know if I've forgotten something.
1. To move the annotations along with domains that move like Katrina.  If we coded these in world coordinates, the annotations would get left behind the domain.  (The box class would probably benefit from this approach so it doesn't get left behind in a moving domain)
2. So that annotations would be drawn around the extents of multiple datasets.  If we stored world coordinates, the annotations would only be drawn around the first dataset loaded in a session, until manually changed.

Reply:
Why would you need a PGroup that upon each update passes a pointer that never changes?

Reply:
> Why would you need a PGroup that upon each update passes a pointer that never changes?

I think that this is a bigger discussion.

This PR is using PWidgets as currently implemented.  Interfacing PWidgets to the ControlExec is a new use case and I think it should be a new Developer Feature.

Stas, if you disagree, then why?  I don't want to list my thoughts and reasons in this thread unless there's a reason we need PWidgets to interface to the ControlExec in this specific PR.

Reply:
@stasj and @sgpearse will attempt a paired-programming effort to resolve